### PR TITLE
Add an option to extend mappings 

### PR DESCRIPTION
**Pull request by V1OL3TF0X**
Add an option to extend mappings that refresh the query result. API to be discussed, that was what felt natural

### CODE REVIEW

**Comment by yochem** [comment]
Thank you for your PR. I would suggest that the new user-configured mapping _overwrites_ the default one (normal mode <CR>). This also gives the user the ability to disable the enter remap. When they want both, they can just add the <CR> mapping too.

When I think about it, another approach would be to expose a usercommand or lua function (e.g. `:JqPlaygroundRunQuery` or `require('jq-playground').run_query` that can be used to add remaps using `vim.keymap.set`. Maybe I like this more, since it might be more expected than a config setting. I also generally do not like it when plugins set all kinds of keymaps, but for this plugin it felt natural to add one (and the original project already did this).

We could keep the config for disabling the default keymap.

What do you think?

**Comment by yochem** [comment]
In the second approach I mentioned, there is no clear way (that I know of) to limit the keymap only to the query buffer. So maybe your approach is the way to go.

**Comment by V1OL3TF0X** [comment]
After some thought, I think I would keep only the run_query_mappings option, to not create too many different options unnecessarily. I really like the idea of the second approach, but I feel like for the right API it would need to be passed through an on_attach option in setup, to then give the user an ability to run it. I feel like that would be a bit overkill for just mappings, and I don't see another usecase, so for just this option I'd stick with a simple table.

**Comment by yochem** [comment]
>  I think I would keep only the run_query_mappings option
Yes, good idea. Maybe name it `query_keymaps`. I think that's clear enough and keymap matches neovims api terms.

**Comment by yochem** [comment]
Should we keep the hint

```jq
# JQ filter: press <CR> in normal mode to execute.
```

in the query buffer?

I also tried to do this with extmarks, but was not satisfied with the result.

**Comment by V1OL3TF0X** [comment]
Decided to add a bit to the comment to hint at the setup options and updated the readme to match the new option.

File: lua/jq-playground.lua [hunk 1]
```diff
@@ -78,7 +78,7 @@ local function create_query_buffer(winopts)
   vim.cmd.startinsert()
 
   vim.api.nvim_buf_set_lines(bufnr, 0, -1, false, {
-    "# JQ filter: press <CR> in normal mode to execute.",
+    "# JQ filter: press set keymap (default <CR> in normal mode) to execute.",
```
Comment (yochem) [inline lua/jq-playground.lua:81]: We can also set this text based on the configured keymap, although that might be a bit overkill

**Review by yochem — APPROVED** [review]
Looks good! Thanks. I'll merge it.

File: lua/jq-playground.lua (hunk 1 above)
Comment (V1OL3TF0X) [reply lua/jq-playground.lua:81]: That's what I thought, especially that we'd have to choose to either generate it for every keymap or just first or... tbh I didn't care enough to xD

**Pull request merged**